File: benchmarks/oopsla25/subjects/094.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

const char* f094(const char* s) {
    if (strlen(s) > 0) {
        char t[2];
        t[0] = s[0];
        t[1] = '\0';
        assert(strcmp(t, "a") == 0);
        assert(strlen(s) == 1);
        return t;
    } else {
        return s;
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f094(my_string);
  return 0;
}

File: benchmarks/oopsla25/subjects/094.py
#!/usr/bin/env python3

import sys

def f094(s: str):
  if len(s) > 0:    
    t = s[0:1]
    assert t == "a"
    assert len(s) == 1
    return t
  else:
    return s
if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f094(my_string)
